invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -339,39 +326,33 @@ public boolean apply(final TrackingRecordId existingRecord) {
+        Preconditions.checkState(isBuilt.get(), "#isContainedIntoExistingUsage(): isBuilt");
         if (existingUsage.isEmpty()) {
             return null;
         }
 
-        final Optional<InvoiceItem> res = Iterables.tryFind(existingUsage, new Predicate<InvoiceItem>() {
-            @Override
-            public boolean apply(final InvoiceItem input) {
-                if (input.getInvoiceItemType() != InvoiceItemType.USAGE) {
-                    return false;
-                }
-                final UsageInvoiceItem usageInput = (UsageInvoiceItem) input;
-                final boolean isContained = usageInput.getUsageName().equals(usage.getName()) &&
-                                            ((startDate.compareTo(usageInput.getStartDate()) >= 0 && endDate.compareTo(usageInput.getEndDate()) < 0) ||
-                                             (startDate.compareTo(usageInput.getStartDate()) > 0 && endDate.compareTo(usageInput.getEndDate()) <= 0));
-                return isContained;
-            }
-        });
-        return res.isPresent() ? res.get() : null;
+        return existingUsage.stream()
+                .filter(input -> {
+                    if (input.getInvoiceItemType() != InvoiceItemType.USAGE) {
+                        return false;
+                    }
+                    final UsageInvoiceItem usageInput = (UsageInvoiceItem) input;
+                    return usageInput.getUsageName().equals(usage.getName()) &&
+                           ((startDate.compareTo(usageInput.getStartDate()) >= 0 && endDate.compareTo(usageInput.getEndDate()) < 0) ||
+                            (startDate.compareTo(usageInput.getStartDate()) > 0 && endDate.compareTo(usageInput.getEndDate()) <= 0));
+                })
+                .findFirst().orElse(null);
     }
 
     protected abstract void populateResults(final LocalDate startDate, final LocalDate endDate, final DateTime catalogEffectiveDate, final BigDecimal billedUsage, final BigDecimal toBeBilledUsage, final UsageInArrearAggregate toBeBilledUsageDetails, final boolean areAllBilledItemsWithDetails, final boolean isPeriodPreviouslyBilled, final List<InvoiceItem> result) throws InvoiceApiException;
 
     protected abstract UsageInArrearAggregate getToBeBilledUsageDetails(final LocalDate startDate, final LocalDate endDate, final List<RolledUpUnit> rolledUpUnits, final Iterable<InvoiceItem> billedItems, final boolean areAllBilledItemsWithDetails) throws CatalogApiException;
 
     private boolean areAllBilledItemsWithDetails(final Iterable<InvoiceItem> billedItems) {
-        boolean atLeastOneItemWithoutDetails = Iterables.any(billedItems, new Predicate<InvoiceItem>() {
-            @Override
-            public boolean apply(final InvoiceItem input) {
-                return input.getItemDetails() == null || input.getItemDetails().isEmpty();
-            }
-        });
-        return !atLeastOneItemWithoutDetails;
+        // FIXME-1615: Use import statement
```

Comment:
Still valid? It doesn't look like `com.google.common.collect.Iterables` is still imported?